core/kernel/notif.c

```diff
@@ -35,6 +51,8 @@ TEE_Result notif_alloc_async_value(uint32_t *val)
 
 	if (!alloc_values_inited) {
 		bit_set(notif_alloc_values, NOTIF_VALUE_DO_BOTTOM_HALF);
+		if (IS_ENABLED(CFG_CORE_IT_NOTIF))
+			bit_set(notif_alloc_values, NOTIF_VALUE_DO_IT);
```

Comment:
This should probably be done unconditionally to make sure it's not found by `bit_ffc()` below.

Reply:
Legacy implementation allows async notif value 1 be used for standard (see linux kernel impl.).

Reply:
I'm sorry, I don't follow. Can you give a pointer?

Reply:
Latest OP-TEE and Linux kernel consider notif value 1 as a generic purpose aysnc notif. So we cannot change Linux to unconditionally consider this value as reserved for "it notif". Therefore, on Linux side, this value is reserved for "it notif" only upon IT_NOTIF capability being enabled. In related Linux kernel patch (https://lore.kernel.org/lkml/20230112145424.3791276-4-etienne.carriere@linaro.org/):
```c
+		else if (optee->smc.sec_caps & OPTEE_SMC_SEC_CAP_IT_NOTIF &&
+			 value == OPTEE_SMC_ASYNC_NOTIF_VALUE_DO_IT)
+			handle_optee_it(optee);
```

I agree we could have the OP-TEE side to always reserve this value 1 for IT_NOTIF, but we cannot do that on Linux side. So for consistency, I bound it to the capabiliy on both side. Do you think it adds confusions?

Reply:
Thanks for the explanation. Yes, I think it's slightly confusing on the OP-TEE side, with a value that is only sometimes reserved. On the Linux side, we have no choice, but on the OP-TEE side, I believe it would be clearer to have it permanently reserved.

Reply:
With latest proposal, there is no more reserved async value related to interrupt notification.

This is at the cost of non-secure world will always invoke (smc fastcall) optee to query pending IRQ after what pending async value (thread related) will will queried.